presentation/backend/src/presentation-backend/RulesetEmbedder.ts

```diff
@@ -33,6 +33,24 @@ export enum DuplicateRulesetHandlingStrategy {
   Replace,
 }
 
+/**
+ * Interface for callbacks which will be called before and after Element/Model updates
+ * @beta
+ */
+interface UpdateCallbacks<TProps> {
+  onBeforeUpdate: (props: TProps) => Promise<void>;
+  onAfterUpdate: (props: TProps) => Promise<void>;
+}
+
+/**
+ * Interface for callbacks which will be called before and after Element/Model is inserted
+ * @beta
+ */
+interface InsertCallbacks<TProps> {
+  onBeforeInsert: (props: TProps) => Promise<void>;
+  onAfterInsert: (props: TProps) => Promise<void>;
+}

```

Comment:
Since we switched from using `ElementProps` and `ModelProps` to a single type - `Entity`, I think there's no need to have there interfaces generic anymore, right?

Reply:
Yep, updated